fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -1110,4 +1112,94 @@ public void testIndexFromIndexRebuild() {
+
+        openSimpleMetaData(hook);
+        buildSrcIndex(srcIndex);
+
+        // Partly build index
+        openSimpleMetaData(hook);
+        buildIndexAndCrashHalfway(tgtIndex, chunkSize, 1, timer,
+                OnlineIndexer.IndexingPolicy.newBuilder()
+                        .setSourceIndex("src_index")
+                        .forbidRecordScan()
+                        .build());
+
+        openSimpleMetaData(hook);
+        String luka = "my name is Luka";
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setIndex(tgtIndex)
+                .build()) {
+            indexer.indexingStamp(OnlineIndexer.IndexingStampOperation.BLOCK, luka, 10);
+        }
+
+        // query, ensure blocked
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setIndex(tgtIndex)
+                .build()) {
+            final AbstractMap<String, IndexBuildProto.IndexBuildIndexingStamp> stampMap =
+                    indexer.indexingStamp(OnlineIndexer.IndexingStampOperation.QUERY, null, 0);
+            System.out.println(stampMap);
+            String indexName = tgtIndex.getName();
+            final IndexBuildProto.IndexBuildIndexingStamp stamp = stampMap.get(indexName);
+            assertEquals(IndexBuildProto.IndexBuildIndexingStamp.Method.BY_INDEX, stamp.getMethod());
+            assertTrue(stamp.getBlock());
+            assertEquals(luka, stamp.getBlockID());
+            assertTrue(stamp.getBlockExpireEpochSeconds() > (System.currentTimeMillis() / 1000));
+            assertTrue(stamp.getBlockExpireEpochSeconds() < 20 + (System.currentTimeMillis() / 1000));
+        }
+
+        // ensure blocked
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = newIndexerBuilder()
+                .addTargetIndex(tgtIndex)
+                .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                        .setSourceIndex("src_index")
+                        .forbidRecordScan()
+                        .build())
+                .setLimit(chunkSize)
+                .setTimer(timer)
+                .build()) {
+            RecordCoreException e = assertThrows(RecordCoreException.class, indexBuilder::buildIndex);
+            assertTrue(e.getMessage().contains("This index was partly built, and blocked"));
+        }
+
+        // Successfully build
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = newIndexerBuilder()
+                .addTargetIndex(tgtIndex)
+                .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                        .setSourceIndex("src_index")
+                        .forbidRecordScan()
+                        .setAllowUnblock(true)
+                        .build())
```

Comment:
Not as part of this PR, but this implies to me that it might be good to just have an option to `unblockAndContinue`, which would get any necessary information for the policy from the stamp.

Reply:
Sound like a good idea!
However, calling indexer without specifying which index to build might be better performed by a layer on top of `OnlineIndexer`. Maybe `StoreIndexer`?

Reply:
Yeah, adding more to `OnlineIndexer` might be excessive, and perhaps something around, or utilities for doing common operations around building indexes would probably be useful. There's been a lot of moderately organic growth in the indexing part of fdb-record-layer, so perhaps revisiting the architecture could result in some better, and hopefully more testable abstractions.